Mini Cart Block contents: Remove item border bottom

danieldudzic opened the pull request:
Remove the item border-bottom, inside of the Mini Cart block contents.

I'm not a huge fan of using `!important` but didn't see a clean way to overwrite the core styles.

<!-- Reference any related issues or PRs here -->
Fixes #5679 

### Screenshots

Before:
<img width="1304" alt="Products_–_kirigami" src="https://user-images.githubusercontent.com/905781/152891863-0d663064-12ae-476b-a13d-9f032a7e91a4.png">

After:
<img width="1307" alt="Products_–_kirigami" src="https://user-images.githubusercontent.com/905781/152891776-c505c811-5a62-4fcd-a6bb-ed545c9885a7.png">

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Active `Storefront` theme and add the Mini Cart Block as a widget.
2. Go on `/shop` page and add a product.
3. Confirm that the flyout mini cart items don't have a border

### User Facing Testing

* [x] Same as above

danieldudzic committed:
Mini Cart: Remove item border bottom #5679

Inside of the Mini Cart block contents, remove the item border bottom

gigitux commented on assets/js/blocks/cart-checkout/mini-cart/style.scss:
```diff
@@ -100,16 +100,25 @@ h2.wc-block-mini-cart__title {
 		margin-right: -$gap;
 		overflow-y: auto;
 		padding-right: $gap;
 
```
Is there a particular reason for adding this style?

I just tried with this code and it seems to work

```suggestion
.wc-block-cart-items__row::after {
    border: none !important
}
```

danieldudzic replied:
Thanks for the review!

The additional code adds a bit of space on top of the mini cart contents further improving it visually.

gigitux replied:
Ok! It makes sense!

gigitux reviewed (commented):
Thanks for working on this! I have just a question about the style that you added.

gigitux reviewed (approved):
LGTM! :shipit: 
NIT: I think that it is a good thing add a comment about why we are using `!important`

danieldudzic committed:
Mini Cart Block contents: Refactor the border fix

Co-authored-by: Luigi Teschio <gigitux@gmail.com>

danieldudzic commented:
The initial fix was overwriting the default border (coming from the cart styles) using `!important`. To remove the bottom border from the mini cart items (without potentially breaking any future CSS overwrites) we have refactored the cart styles (and removed the `!important` overwrite). Can you please have a look?

gigitux commented:
> The initial fix was overwriting the default border (coming from the cart styles) using `!important`. To remove the bottom border from the mini cart items (without potentially breaking any future CSS overwrites) we have refactored the cart styles (and removed the `!important` overwrite). Can you please have a look?

Thanks for your refactor! LGTM! 

Let's wait if Rubik folks have feedback! 👍

ralucaStan commented on assets/js/blocks/cart-checkout/cart/style.scss:
```diff
@@ -265,6 +264,18 @@ table.wc-block-cart-items {
+// https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5679
+.is-medium,
+.is-small,
+.is-mobile:not(.wc-block-mini-cart__drawer) {
```
This solution works but it has the disadvantage of not using a targeted class for the Mini Cart (is-mobile is a class that is also used in other places), and it's also quite far from the line that it is overwriting. 

My recommendation is to add a class to the table used in the MiniCart so we can better target its rows.
- Add this under [this line](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/8475bb8c47499ad535aceb4391d9bafa30ec3228/assets/js/blocks/cart-checkout/mini-cart-contents/inner-blocks/mini-cart-products-table-block/block.tsx#L17) `className="wc-block-mini-cart-items"`
- Add the selected rule from the image below and remove the original line with the border
<img width="683" alt="Screenshot 2022-02-14 at 18 25 32" src="https://user-images.githubusercontent.com/1628454/153914922-e4a7216b-a245-47b6-9dc9-9cd00c4dcd02.png">

Notice that this way you don't need the comments explaining what is happening.

```
		&:not(.wc-block-mini-cart-items) {
			.wc-block-cart-items__row {
				@include with-translucent-border( 0 0 1px );
			}
		}
```

ralucaStan reviewed (changes requested):
I've made a suggestion, let me know what you think.

danieldudzic committed:
Refactor the mini cart item border bottom fix

ralucaStan reviewed (approved):
Works as expected. Ready to ship. 🚢

danieldudzic commented:
> Works as expected. Ready to ship. 🚢

Thank you for the suggestion (valid points) and the review!
